src/sentry/api/serializers/models/rule.py

```diff
@@ -289,3 +302,192 @@ def serialize(self, obj, attrs, user, **kwargs) -> RuleSerializerResponse:
+            result[workflow]["action_match"] = (
+                workflow.when_condition_group.logic_type if workflow.when_condition_group else None
+            )
+            result[workflow]["filter_match"] = workflow_to_filter_match.get(workflow, "none")
+            result[workflow]["rule_id"] = workflow_rule_ids[workflow.id]
+
+            # Generate conditions and filters
+            all_conditions: list[dict[str, Any]] = []
+            all_filters: list[dict[str, Any]] = []
+
+            def update_condition_name(condition: dict[str, Any]) -> dict[str, Any]:
+                condition["name"] = generate_rule_label(
+                    project=result[workflow]["projects"][0], rule=None, data=condition
+                )
+                return condition
+
+            def generate_condition_filters(conditions: list[DataCondition], is_filter: bool):
+                for cond in conditions:
+                    condition, filters = translate_to_rule_condition_filters(
+                        cond, is_filter=is_filter
+                    )
+                    if condition:
+                        all_conditions.append(update_condition_name(condition))
+                    all_filters.extend([update_condition_name(f) for f in filters])
+
+            trigger_conditions = (
+                list(workflow.when_condition_group.conditions.all())
+                if workflow.when_condition_group
+                else []
+            )
+            generate_condition_filters(trigger_conditions, is_filter=False)
+            filter_conditions = workflow_to_filters[workflow]
+            generate_condition_filters(filter_conditions, is_filter=True)
+
+            result[workflow]["conditions"] = all_conditions
+            result[workflow]["filters"] = all_filters
+
+        if "lastTriggered" in self.expand:
+            result_qs = reduce(
+                lambda q1, q2: q1.union(q2),
+                [
+                    WorkflowFireHistory.objects.filter(workflow=item)
+                    .order_by("-date_added")
+                    .values("workflow_id", "date_added")[:1]
+                    for item in item_list
+                ],
+            )
+            last_triggered_lookup = {wfh["workflow_id"]: wfh["date_added"] for wfh in result_qs}
+
+            for item in item_list:
+                result[item]["last_triggered"] = last_triggered_lookup.get(item.id, None)
+
+        return result
+
+    def serialize(self, obj: Workflow, attrs, user, **kwargs) -> RuleSerializerResponse:
+        environment = attrs["environment"]
+
+        action_match = attrs["action_match"]
+        if action_match == DataConditionGroup.Type.ANY_SHORT_CIRCUIT:
+            action_match = "any"
+
+        filter_match = attrs["filter_match"]
+        if filter_match == DataConditionGroup.Type.ANY_SHORT_CIRCUIT:
+            filter_match = "any"
+
+        d: RuleSerializerResponse = {
```

Comment:
`d`? is this short for `dictionary`? -- could we call this `rule` or something instead?